### ISSUE TITLE
Attributes not visible in Home Assistant

### ISSUE THREAD

**Issue by allgrinder** [gdoor-org/gdoor#51]
Hey,
unfortunately, I can't retrieve the BUS-ID as described in the documentation, so I can't control the door via Home Assistant. What am I doing wrong?

Best regards,

![grafik](https://github.com/user-attachments/assets/f0884027-b585-4947-bec3-a2c5d2888890)

**Comment by m5ck** [comment]
Hi @allgrinder,

Can you give a bit more details on what you did step by step? Did you see any BUS messages?

**Comment by allgrinder** [comment]
Ah sorry, that was my mistake.
In the end, I simply activated and tested the video function or door opening on the indoor unit. Gdoor was integrated via MQTT...
I also received a corresponding entry in the logbook. Some of them are duplicated, but that’s not a big issue for now... I’ll look into that next.
According to the documentation: https://gdoor-org.github.io/documentation/homeassistant.html, the result in the logbook should look like this:

GDoor changed to {"action": "BUTTON_RING", "parameters": "0360", "source": "A286FD", "destination": "000000", "type": "OUTDOOR", "busdata": "011011A286FD0360A04A"}

Apparently, the parameter is crucial... but I’m not receiving any parameter.

Also, the examples for typical use cases are not working either... I keep getting the message "shell command not found," which is surprising to me.

**Comment by DaSchaef** [comment]
I can not comment on HA (I'm a OpenHab user ^^).

> Apparently, the parameter is crucial... but I’m not receiving any parameter.

You received them :-)

> {"action": "BUTTON_RING", "parameters": "0360", "source": "A286FD", "destination": "000000", "type": "OUTDOOR", "busdata": "011011A286FD0360A04A"}

"parameters" =  "0360"

The parameters field has a different meaning for different bus actions.
For ring button pressed ("BUTTON_RING"), it mostly codes which button on the outdoor station was pressed, if there are e.g. multiple stations in the house. Therefore it is important for you, to use the value as filter, to not react to all buttons :)

**Comment by DaSchaef** [comment]
If you do not receive anything, you may want to try http://mqtt-explorer.com/ to "listen" to the GDoor ---> HomeAssistent communication. So we can avoid that HA is hiding the GDoor message from you somehow.

**Comment by allgrinder** [comment]
No, that was the example from the GitHub page. I didn’t receive anything – just 0000.
![IMG_0337](https://github.com/user-attachments/assets/927f2e4c-52b1-4653-850f-a906f4856bc8)

**Comment by DaSchaef** [comment]
This may indicate that you do not receive any events? The "IDLE" is from GDoor itself and only a kind of keepalive ("I'm here") packet.

You should receive events, when you do something with the bus e.g. press the door button on the outdoor station. For testing, you can also use the light button on the indoor station. This is often easier to reach ;)

If you do not receive any bus message, we need to investigate further:
- Is GDoor connected to the bus?
- Can you listen to the raw MQTT messages (e.g. http://mqtt-explorer.com/) so that we can take HomeAssistant out of the possible error sources
- You should see MQTT messages on the topic, also set up in HA in the MQTT Explorer.

**Comment by allgrinder** [comment]
Yes, I do get all the events displayed correctly. But they just come without parameters.
![IMG_0340](https://github.com/user-attachments/assets/8cf37c2e-3801-4d73-84ad-0734950d415c)

**Comment by DaSchaef** [comment]
Hm, I'm no HA user. Can you click on it and get more details, (meaning the raw message)?
I think HA is hiding it from you :)

Otherwise we need another HA user here to support a bit - but seems solvable :+1:

**Comment by allgrinder** [comment]
All good! Thanks for your support. My post with the screenshot at 17:14 shows the details of a Button Ring. Once I’m home and have some time, I’ll install the Explorer.

**Comment by DaSchaef** [comment]
The screenshot at 17:14 is an IDLE message, no BUTTON_RING ? This is why the parameter is empty in this mentioned image :)

**Comment by allgrinder** [comment]
Yes, that confuses me as well… I click on the corresponding event, and it only shows Idle.

**Comment by allgrinder** [comment]
It's really strange – MQTT Explorer shows the parameter, and Home Assistant also displays the various statuses. But the bus adapter's log continuously shows 'Idle' or doesn't indicate that another status briefly appeared.

![grafik](https://github.com/user-attachments/assets/63d0b42c-7a97-4bb5-8f37-09a147c1f75d)

![grafik](https://github.com/user-attachments/assets/779d07b3-53a2-4947-9933-ac41f63e2a9b)

**Comment by allgrinder** [comment]
The ticket can be closed. I can control the door via MQTT and receive a notification when the doorbell rings. That was my goal, and it works! Thank you for your help!

**Comment by jschroeter** [comment]
Did you check the [Logbook](https://www.home-assistant.io/integrations/logbook/) in HA? Its enabled by default and usually doesn’t need to be configured. There you should see all events including details.
In the History view you’ll mostly only see the details of the last event, which is always BUS_IDLE.

**Comment by jschroeter** [comment]
Reopening this as we indeed have 2 problems here - I didn't notice them so far as I was mainly using the serial integration:

1. due to [this line](https://github.com/gdoor-org/gdoor/blob/feb611d1c08d6f1e54a5dbc9a58bdd9572465d75/firmware/esp32/gdoor/src/mqtt_helper.cpp#L132), HA only shows the `action` in the Logbook. Unfortunately HA only allows to see the JSON attributes for the latest state, which is always `BUS_IDLE` (as we send this action after every other action) and therefore you can't see e.g. the `parameters` of an relevant action.
  <img width="907" src="https://github.com/user-attachments/assets/c3856fa6-49de-4894-a85a-f67caab59ef7" />

2. quite frequently HA reports that the sensor is unavailable/unknown and actions are missing on the HA sensor (but they do appear when listening to the MQTT topic directly). I'm pretty sure this happens because HA only allows 255 characters for a value and we often exceed this.

Suggestions:
- to fix 1),
  - A) we change mqtt_helper.cpp#L132 to `"value_template": "{{ value }}"` which would display the whole JSON as string. Works but looks a bit clunky, especially in the history view.
  <img width="664" src="https://github.com/user-attachments/assets/68270cc5-813e-4d74-9c42-6d4bd9b49c62" />
  <img width="212" src="https://github.com/user-attachments/assets/fcf1a6c9-7e61-4ced-b48d-9c9fc11fad21" />

  - B) or to `"value_template": "{{ value_json.action }} {{ value_json.source }} {{ value_json.destination }} {{ value_json.parameters }}"`. Looks cleaner but needs explanation what these values actually are. Nice side effect: since the event id is not included, equal actions are shown with the same color code.
  <img width="666" src="https://github.com/user-attachments/assets/251cf316-219b-4aee-a960-a6697fdb078e" />
  <img width="214" src="https://github.com/user-attachments/assets/1b7a7df4-aaf2-4f11-9121-b819ab07df32" />

  - C) we drop the BUS_IDLE action again, I personally still don't like it and so far don't have a use case for it [as discussed before](https://github.com/gdoor-org/gdoor/issues/27), but I think some find it useful.

- to fix 2)
  - D) we remove the `raw` attribute as it duplicates `busdata` anyway. By that we should stay under 255 chars.
  - E) we move it and maybe other debug-related attributes (`valid`, newly added `busvoltage`) to an own MQTT topic, e.g. `gdoor/debug` or `gdoor/rx_debug`

Any opinions? I'd vote for C) and/or B) + E).

**Comment by allgrinder** [comment]
Yes, @jschroeter is right... it is indeed a bit confusing. But once you're aware of the peculiarity, you can deal with it. I can control the door, receive signals, and perform actions... so I'm very satisfied :)

But C is probably pragmatic.

**Comment by DaSchaef** [comment]
Just to keep ypu updated, I'm currently waiting to have a bit more time, I want to first re-iterated the current audio hw with its bugs, as we need a new firmware for this.
Therefore I'm a bit slow on the firmware side, as I do not want to re-implement twice later on :-)

If there is something more critical, please just remind me, or highlight the importance :)

**Comment by struhs** [comment]
all parameters of the last 10 day (default?) can be fetched from the recorder. Unfortunately there is no simple UI for that - afaik.
You can add the SQlite - Addon and try the SQL command
`SELECT    s.state,    sa.shared_attrs,   datetime(s.last_updated_ts, 'unixepoch', 'localtime') AS lokales_datum FROM states s LEFT JOIN state_attributes sa ON s.attributes_id = sa.attributes_id WHERE s.metadata_id = "2848" AND s.state != "BUS_IDLE"  ORDER BY s.last_updated_ts DESC`
This should show you all non BUS_IDLE entries.
